app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,14 @@
+class PiecesController < ApplicationController
+  def update
+    @piece = Pieces.find(params[:id])
+    new_x = params[:x].to_i
+    new_y = params[:y].to_i
+    return update_attributes(x_position: new_x, y_position: new_y) if @piece.move!(new_x, new_y)
```

Comment:
It actually might make sense to call `update_attributes` in a controller if that's the only thing you're doing, since the `update_attributes` itself would be the "behavior" you want to happen in the model.  But in our case, we have a more complex behavior (`move!`) that we're calling that will do more than just update the attributes.